src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -1204,10 +1283,226 @@ class Simplex_tree {
+      auto res_ins = root_.members().emplace(u, Node(&root_,fil));
+      if (res_ins.second) { //if the vertex is not in the complex, insert it
+        added_simplices.push_back(res_ins.first); //no more insert in root_.members()
+        update_simplex_tree_after_node_insertion(res_ins.first);
+      } else {
+        GUDHI_CHECK(false,"Simplex_tree::insert_edge_as_flag - insert a vertex already in the complex");
+      }
+      return; //because the vertex is isolated, no more insertions.
+    }
+    // else, we are inserting an edge: ensure that u < v
+    if (v < u) { std::swap(u,v); }
+
+    //Note that we copy Simplex_handle (aka map iterators) in added_simplices
+    //while we are still modifying the Simplex_tree. Insertions in siblings may
+    //invalidate Simplex_handles; we take care of this fact by first doing all
+    //insertion in a Sibling, then inserting all handles in added_simplices.
+
+    //check whether vertices u and v are in the tree. If not, return an error.
+    auto sh_u = root_.members().find(u);
+    GUDHI_CHECK(sh_u != root_.members().end() &&
+          root_.members().find(v) != root_.members().end(),
+          std::invalid_argument(
+                  "Simplex_tree::insert_edge_as_flag - insert an edge whose vertices are not in the complex")
+                );
+
+    //check if the edge {u,v} is already in the complex, if true, nothing to do.
+    if (has_children(sh_u) &&
+        sh_u->second.children()->members().find(v) != sh_u->second.children()->members().end())
+    {
+      GUDHI_CHECK(false,"Simplex_tree::insert_edge_as_flag - insert an edge already in the complex");
+      return;
+    }
+
+    //upper bound on dimension
+    dimension_ = dim_max;
+    dimension_to_be_lowered_ = true;
+
+    //for all siblings containing a Node labeled with u (including the root), run
+    //compute_punctual_expansion
+    //todo parallelise
+    auto ptr_list_u = nodes_by_label_.find(u);//all Nodes with u label
+
+    GUDHI_CHECK(ptr_list_u != nullptr,"Simplex_tree::insert_edge_as_flag - cannot find the list of Nodes with label u");
+
+    for (auto hook_u_it = ptr_list_u->begin(); hook_u_it != ptr_list_u->end(); ++hook_u_it)
+    {
+      Node & node_u = static_cast<Node&>(*hook_u_it); //corresponding node
+      Siblings * sib_u = self_siblings(node_u, u);    //Siblings containing node
+      if (sib_u->members().find(v) != sib_u->members().end()) {
+        int curr_dim = dimension(node_u,u);
+        if (curr_dim < dim_max){
+          if (!has_children(node_u, u)) { //now has a new child Node labeled v
+            node_u.assign_children(new Siblings(sib_u, u));
+          }
+          compute_punctual_expansion(
+                v,
+                node_u.children(),
+                fil,
+                dim_max - curr_dim - 1, //>= 0
+                added_simplices );      //u on top
+        }
+      }
+    }
+    //sort added_simplices appropriately, using reverse_lex_order (all new simplices have
```

Comment:
Better remove this code that is commented out, it is confusing.

Reply:
Sorry, I used it while testing and then forgot about it. It is removed.